src/SoftwareSerial.h

```diff
@@ -120,22 +188,26 @@ class SoftwareSerial : public Stream {
 	bool m_invert;
 	bool m_overflow = false;
 	int8_t m_dataBits;
+	int8_t m_parityBits;
+    int8_t m_stopBits;
+    ParityMode m_parity;
 	int32_t m_bitCycles;
 	uint32_t m_periodDeadline;
 	bool m_intTxEnabled;
 	int m_inPos, m_outPos;
 	int m_bufSize = 0;
 	uint8_t *m_buffer = 0;
+	uint8_t *m_pbuffer = 0;		// buffer to store parity for received chars
```

Comment:
As soon as you rebase to more current version of EspSoftwareSerial, there's the circular_queue instead of a plain memory block. Instantiating that class template with a struct { uint8_t data; bool parity; } is easy - but unconditionally uses twice as much RAM...

Reply:
Looking forward to the circular buffer but I have no clue how bad it would be to block the extra RAM. 
Would it be worthwhile/possible to allocate the buffer on demand, only if parity is used?
I've played with idea to make the support for parity optional using #defines. Most use cases will very likely be 8N1.